src/main/java/upb/dice/twitter/TweetSearchApplication.java

```diff
@@ -1,88 +1,153 @@
+                            iterator1 = currentLat.iterator();
+                            iterator2 = currentLon.iterator();
+                            iterator3 = currentRad.iterator();
+
+                            //runs forever with the queues being assigned inside the while loop
+                            while ((iterator.hasNext()) || (iterator1.hasNext() && iterator2.hasNext() && iterator3.hasNext())) {
+
+                                //start keyword based search
+                                while (iterator.hasNext()) {
+                                    String keyCurrent = currentKey.peek();
+                                    LOGGER.info("Current keyword is: " + keyCurrent);
+
+                                    try {
+                                        Query keywordQuery = tweetExplorer.keywordQueryGen(keyCurrent);
+                                        tweetExtractor.getTweet(keywordQuery);//Start the search
+                                        currentKey.remove(); //Remove from the current queue
+                                    } catch (IOException | InterruptedException ex) {
+                                        ex.printStackTrace();
+                                    }
+                                }
+
+                                //start location based search
+                                while (iterator1.hasNext() && iterator2.hasNext() && iterator3.hasNext()) {
+                                    Double lat = currentLat.peek();
+                                    Double lon = currentLon.peek();
+                                    Double rad = currentRad.peek();
+                                    LOGGER.info("Current Latitude, Longitude and Radius: " + lat.toString() + "," + lon.toString() + ',' + rad.toString());
+                                    try {
+                                        Query locationQuery = tweetExplorer.locationQueryGen(lat, lon, rad);
+                                        tweetExtractor.getTweet(locationQuery);//Search start
+                                        //remove from the current queue
+                                        currentLat.remove();
+                                        currentLon.remove();
+                                        currentRad.remove();
+
+                                    } catch (IOException | InterruptedException ex) {
+                                        ex.printStackTrace();
+                                    }
+                                }
+                                //initialize new queue by copying values from the original queue
+                                currentKey = new LinkedList<>(keywordQueue);
+                                currentLat = new LinkedList<>(latitudeQeue);
+                                currentLon = new LinkedList<>(longitudeQeue);
+                                currentRad = new LinkedList<>(radiusQeue);
```

Comment:
This is not how we discussed it, is it? Because at the moment, the client will _always_ start with the same queries regardless what happened in the previous run of the timer.

However, the idea of having different queries, etc. was that we can make sure that if a query was the last during a previous run and hit the limit, it will be at the end of the queue for the next run.